### PR TITLE
Put third party crates behind feature flags

### PR DESCRIPTION
**Pull request by tobiemh**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,18 +16,28 @@ categories = ["encoding"]
...
 
+[features]
+default = []
+chrono = ["dep:chrono"]
+geo = ["dep:geo"]
+ordered-float = ["dep:ordered-float"]
+regex = ["dep:regex"]
+roaring = ["dep:roaring"]
+rust_decimal = ["dep:rust_decimal"]
+uuid = ["dep:uuid"]
```
Comment (rushmorem) [inline Cargo.toml]: I don't think it's necessary to redefine the feature flags if we are reusing the crate name as-is. These are already implied by `optional = true`.

File: src/implementations/chrono.rs [hunk 2]
```diff
@@ -1,6 +1,8 @@
+#![cfg(feature = "chrono")]
```
Comment (rushmorem) [inline src/implementations/chrono.rs:1]: Shouldn't we put these on module definitions instead? What happens when the feature is not enabled? Won't the `mod` statement break?

Comment (tobiemh) [reply src/implementations/chrono.rs:1]: It doesn't appear to @rushmorem ... !

File: Cargo.toml (hunk 1 above)
Comment (tobiemh) [reply Cargo.toml]: Ah ok cool. How would a user specify that they want the `uuid` dependency feature enabled in their Cargo.toml file?

Comment (tobiemh) [reply Cargo.toml]: Ok, just removed these explicit features ✅

**Review by rushmorem — APPROVED** [review]

**Pull request merged**